### PR TITLE
Adjust to OTSpanContext changes and "cleanse"

### PR DESCRIPTION
**Pull request by bhs**
In this case, "cleanse" involves a bunch of Pods work to make the example reflect the reality of what a user of this pod will do. Also, wherever possible, LS\* has been replaced by OT*. This includes the singleton, which should be an OT thing, not an LS thing.

As with other SpanContext changes, this PR cleans up some subtle cases in extract and leads to better separation of concerns around causal relationships between spans.

### CODE REVIEW

File: Pod/Classes/LSSpan.m [hunk 1]
```diff
@@ -232,26 +192,6 @@ - (void) finishWithTime:(NSDate *)finishTime {
     [m_tracer _appendSpanRecord:record];
 }
 
-- (NSString*)traceGUID {
```
Comment (bhs) [inline Pod/Classes/LSSpan.m:235]: (these were not used anywhere)

File: Pod/Classes/LSTracer.h [hunk 2]
```diff
@@ -1,116 +1,82 @@
...
-#pragma mark - Shared instance initialization
+#pragma mark - LSTracer initialization
 
 /**
- * @see `+[LSTracer initSharedTracer:groupName:hostport]` for parameter details.
+ * @see `-[LSTracer initWithToken:componentName:hostport]` for parameter details.
  *
  * @return An `LSTracer` instance that's ready to create spans and logs.
  */
-+ (instancetype) initSharedTracer:(NSString*)accessToken;
```
Comment (bhs) [inline Pod/Classes/LSTracer.h:25]: LSTracer no longer does the singleton work... `OTGlobal` is the place for that.

File: examples/LightStepTestUI/Pods/lightstep/LICENSE [hunk 3]
```diff
@@ -0,0 +1,19 @@
+Copyright LightStep (http://lightstep.com/)
```
Comment (bhs) [inline examples/LightStepTestUI/Pods/lightstep/LICENSE:1]: I'm truly not sure whether the example's Pod deps should be checked in or not... checking them in makes the example-builds easier (which is why I've done it), but also creates duplication.

File: examples/LightStepTestUI/LightStepTestUI/AppDelegate.m [hunk 4]
```diff
@@ -14,11 +15,11 @@ @implementation AppDelegate
 
 
 - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
-    [LSTracer initSharedTracer:@"{your_access_token}"];
-
-    [LSTracer sharedTracer].flushIntervalSeconds = 5;
-    [LSTracer sharedTracer].maxLogRecords = 600;
-    [LSTracer sharedTracer].maxSpanRecords = 600;
+    LSTracer* tracer = [[LSTracer alloc] initWithToken:@"DEVELOPMENT_TOKEN_bhs"];
```
Comment (bcronin) [inline examples/LightStepTestUI/LightStepTestUI/AppDelegate.m]: Should change this back to `{your_access_token}`.

File: examples/LightStepTestUI/Podfile [hunk 5]
```diff
@@ -2,13 +2,14 @@
 # platform :ios, '8.0'
 # Uncomment this line if you're using Swift
 # use_frameworks!
+#
+source 'https://github.com/bensigelman/tmp-pods.git'
```
Comment (bcronin) [inline examples/LightStepTestUI/Podfile]: Should this be checked in?

File: examples/LightStepTestUI/Pods/Target Support Files/lightstep/lightstep-dummy.m [hunk 6]
```diff
@@ -0,0 +1,5 @@
+#import <Foundation/Foundation.h>
+@interface PodsDummy_lightstep : NSObject
+@end
+@implementation PodsDummy_lightstep
+@end
```
Comment (bcronin) [inline examples/LightStepTestUI/Pods/Target Support Files/lightstep/lightstep-dummy.m:5]: I don't know what this file is, but should this be checked in?

**Comment by bcronin** [comment]
As you noted, this is pretty hard to review given all the Pods related diffs :)

I added some comments on things that I'm not sure should be checked in. Other than that, I didn't notice anything that should get in the way of merging this.

File: examples/LightStepTestUI/Pods/Target Support Files/lightstep/lightstep-dummy.m (hunk 6 above)
Comment (bhs) [inline examples/LightStepTestUI/Pods/Target Support Files/lightstep/lightstep-dummy.m:5]: It's something that `pod` creates -- I'll leave it in given my lack of understanding.

File: examples/LightStepTestUI/Podfile (hunk 5 above)
Comment (bhs) [inline examples/LightStepTestUI/Podfile]: NO. Thanks.

**Pull request merged**